index.d.ts

```diff
@@ -0,0 +1,280 @@
+     * @param {string} tableName
+     * @param {string} description
+     * @param columnDefinitions
+     * @returns {Promise<string>}
+     */
+    createTable(appSlug: string, tableName: string, description: string, columnDefinitions: ColumnDefinitionInput[]): Promise<string>
+
+    /**
+     * Ensures that a table exists before trying trying to create.
+     *
+     * @param {string} appSlug
+     * @param {string} tableName
+     * @param {string} description
+     * @param columnDefinitions
+     * @returns {Promise<string>}
+     */
+    ensure(appSlug: string, tableName: string, description: string, columnDefinitions?: ColumnDefinitionInput[]): Promise<string>
+  }
+
+  interface ColumnDefinitionInput {
+    input: ColumnDefinition
+  }
+
+  /**
+   * The different types of column definitions.
+   */
+  type ColumnDefinition =
+    DateColumnDefinition | DateTimeColumnDefinition | EmailColummDefinition | NumberColumnDefinition | TextColumnDefinition;
+
+  /**
+   * The string literals for column types.
+   */
+  type ColumnType = "DATE" | "DATETIME" | "EMAIL" | "NUMBER" | "TEXT";
+
+  /**
+   * Common properties on all columns
+   */
+  interface GenericColumnDefinition<T extends ColumnType> {
+    columnType: T;
+    name: string;
+    description: string;
+    nullabe?: string;
+  }
+
+  /**
+   * Represents DATE columns
+   */
+  interface DateColumnDefinition extends GenericColumnDefinition<"DATE"> {
+  }
+
+  /**
+   * Represents DATETIME columns
+   */
+  interface DateTimeColumnDefinition extends GenericColumnDefinition<"DATETIME"> {
+  }
+
+  /**
+   * Represents EMAIL columns
+   */
+  interface EmailColummDefinition extends GenericColumnDefinition<"EMAIL"> {
+  }
+
+  /**
+   * Represents NUMBER columns
+   */
+  interface NumberColumnDefinition extends GenericColumnDefinition<"NUMBER"> {
+    numberType: NumberType;
+  }
+
+  /**
+   * The type of number to represent
+   */
+  type NumberType = "INTEGER" | "FLOAT";
+
+  /**
+   * Reprents TEXT columns
+   */
+  interface TextColumnDefinition extends GenericColumnDefinition<"TEXT"> {
+    textLength?: number;
+    textEncrypted?: boolean;
+  }
+
+  interface GraphQL {
+
+    /**
+     * Runs a query against the GraphQL API
+     *
+     * @param {string} query
+     * @param {any[]} variable
+     * @returns {Promise<any>}
+     */
+    query(query: string, variable: any[]): Promise<any>;
+
+  }
+
+  /**
+   * The request object passed into the function
+   */
+  interface Request extends e.Request {
+    ctx: Context
+  }
+
+  /**
+   * The response object passed into the function
+   */
+  interface Response extends e.Response {
+  }
+
+  /**
+   * The function callback is passed into every function request.
+   *
+   * If an error is present, then it has already been sent to the client so the response
+   * object cannot be mutated.
+   */
+  type FunctionCallback =  (err: any | undefined, context: Context) => void;
+
+  type JwtToken = string | object;
```

Comment:
Looking at the current usage the `ctx.token` is an object with at least `username` property.
`string | object` -> `{ username: string }`